assets/src/state.ts

```diff
@@ -40,6 +43,28 @@ export enum OpenView {
   Late,
 }
 
+interface CreatePresetModal {
+  type: "CREATE_PRESET"
+  createCallback: (arg0: string, arg1: React.Dispatch<Action>) => void
+}
+
+interface SavePresetModal {
+  type: "SAVE_PRESET"
+  saveCallback: (arg0: React.Dispatch<Action>) => void
+  presetName: string
+}
+
+interface DeletePresetModal {
+  type: "DELETE_PRESET"
+  deleteCallback: (arg1: React.Dispatch<Action>) => void
```

Comment:
Hmm, I do see the issue of serializability being important, especially as Skate gets more complex and we might want to hold onto more of the user's state if they close the tab, get logged out, etc. This would be fairly straightforward for the delete case, but then in the save or create case that moves some of the logic about whether to save or create into the modal element, which feels weird. One way of getting around that could be to simply store the action type ("save" or "create") as an enum type on the `SaveOrCreatePresetModal` interface. I'm torn as to how much better that would be, though, especially if we haven't actually decided that we want to serialize the state.

Reply:
it looks like the only data you'd need to save is the uuid and sometimes the preset name, and there's no logic besides choosing the action.

```
          saveCallback: (dispatch: React.Dispatch<Action>) => {
            dispatch(savePreset(action.payload.routeTab.uuid))
          },

          createCallback: (
            presetName: string,
            dispatch: React.Dispatch<Action>
          ) => {
            dispatch(createPreset(action.payload.routeTab.uuid, presetName))
          },

        deleteCallback: (dispatch: React.Dispatch<Action>) => {
          dispatch(deletePreset(action.payload.routeTab.uuid))
        },
```

Reply:
Right, but I specifically wanted to avoid having the logic around choosing an action in the component, instead letting it live in either the state reducer or the model. The solution I outlined with adding an `actionType` parameter is one option, but that still involves having at least some action dispatch logic going on in the component itself. I think if we do want to serialize the state this can be refactored but as it is it seems like it would make the code slightly more difficult to follow without a clear benefit.